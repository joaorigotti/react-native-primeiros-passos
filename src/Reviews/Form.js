import React, { Component } from 'react';
import { 
  StyleSheet,
  TextInput,
  Text,
  View,
  TouchableOpacity,
} from 'react-native';
import { getWidth } from '../Utils';
import Title from '../Title';

class Form extends Component {
  constructor(props) {
    super(props);
    this.state = { 
      name: '',
      review: '',
    };
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  _resetForm() {
    this.setState(() => ({ name: '', review: ''}));
  }

  handleSubmit() {
    const review = {
      name: this.state.name,
      review: this.state.review,
    };

    if (review.name !== '' && review.review !== '') {
      this.props.action(review, this.props.productId);
      this._resetForm();
    }
    
  }

  render() {
    return (
      <View style={styles.container}>
        <Title size={21}>Deixe sua review</Title>
        <TextInput
          style={styles.input}
          onChangeText={(name) => this.setState({name})}
          value={this.state.name}
          placeholder="nome"
          underlineColorAndroid="transparent"
        />
        <TextInput
          multiline={true}
          style={styles.inputMultiline}
          onChangeText={(review) => this.setState({review})}
          value={this.state.review}
          placeholder="review"
          underlineColorAndroid="transparent"
        />
        <TouchableOpacity onPress={this.handleSubmit} style={styles.submitContainer}>
          <Text style={styles.submit}>ENVIAR</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

const input = {
  height: 40,
  borderLeftWidth: 2,
  borderLeftColor: '#b3b3b3',
  borderBottomWidth: 1,
  borderBottomColor: '#e6e6e6',
  fontFamily: 'RobotoSlab-Regular',
  fontSize: 12,
  paddingLeft: 5,
  marginTop: 10,
};

const styles = StyleSheet.create({
  container: {
    width: getWidth(),
    paddingLeft: 20,
    paddingRight: 20,
    marginTop: 40,
    marginBottom: 40,
  },

  input,

  inputMultiline: Object.assign({}, input, { 
    marginTop: 0,
    height: 120,
  }),

  submitContainer: {
    width: 80,
    alignSelf: 'flex-end'
  },

  submit: {
    fontFamily: 'RobotoSlab-Regular',
    fontSize: 18,
    color: '#4d4d4d',
    marginTop: 10,
    alignSelf: 'flex-end'
  }
});

export default Form