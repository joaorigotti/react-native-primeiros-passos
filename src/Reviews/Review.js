import React from 'react';
import {
  StyleSheet,
  View,
  Text,
} from 'react-native';
import { getWidth } from '../Utils';
import Title from '../Title';

const Review = props => {
  return (
    <View>
      <View style={styles.container}>
        <Title size={16} bold>{props.name}</Title>
        <Text style={styles.review}>{props.text}</Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    width: getWidth(),
    paddingLeft: 20,
    paddingRight: 20,
    paddingTop: 10,
    paddingBottom: 10,
  },

  review: {
    fontFamily: 'RobotoSlab-Regular',
    fontSize: 14,
    color: '#4d4d4d',
    marginTop: 5,
  }
});

export default Review