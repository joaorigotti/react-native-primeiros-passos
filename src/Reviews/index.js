import React from 'react';
import {
  View,
  FlatList,
} from 'react-native';
import Review from '../Review';
import Form from '../Form';
import Separator from '../Separator';

const Reviews = props => {
  const { productId, action } = props.form;

  return (
    <View>
      {/* <FlatList
        data={props.value}
        keyExtractor={item => item.id}
        renderItem={
          ({item}) => <Review name={item.name} text={item.review} />
        }
      /> */}
      {
        props.value.map(item => {
          return <Review key={item.id} name={item.name} text={item.review} />
        })
      }
      <Form productId={productId} action={action} />
    </View>
  );
}

export default Reviews
