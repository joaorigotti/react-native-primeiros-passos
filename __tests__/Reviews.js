import React from 'react';
import { shallow } from 'enzyme';
import Reviews from '../src/Reviews';

describe('<Reviews />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Reviews value={[]} form={{productId:1,action: ()=>{}}} />
    );
    
    expect(wrapper).toMatchSnapshot();
  });
});