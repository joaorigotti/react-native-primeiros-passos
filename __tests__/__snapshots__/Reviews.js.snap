// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Reviews /> renders correctly 1`] = `
<View>
  <Form
    action={[Function]}
    productId={1}
  />
</View>
`;
